src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,433 @@
+
+            // 3. Find the literal arguments, and the mapping parameter will be replaced by that literal expression
+            // Example:
+            // Before:
+            // void Caller(int k)
+            // {
+            //     Callee(1, k);
+            // }
+            // void Callee(int i, int j)
+            // {
+            //     DoSomething(i, k);
+            // }
+            // After:
+            // void Caller(int k)
+            // {
+            //     DoSomething(1, k);
+            // }
+            // void Callee(int i, int j)
+            // {
+            //     DoSomething(i, j);
+            // }
+            var operationsWithLiteralArgument = allArgumentOperations
+                .WhereAsArray(argument =>
+                    _syntaxFacts.IsLiteralExpression(argument.Children.Single().Syntax) && argument.ArgumentKind == ArgumentKind.Explicit);
+
+            // 4. Find the default value parameters. Similarly to 3, they should be replaced by the default value.
+            // Example:
+            // Before:
+            // void Caller(int k)
+            // {
+            //     Callee();
+            // }
+            // void Callee(int i = 1, int j = 2)
+            // {
+            //     DoSomething(i, k);
+            // }
+            // After:
+            // void Caller(int k)
+            // {
+            //     DoSomething(1, 2);
+            // }
+            // void Callee(int i = 1, int j = 2)
+            // {
+            //     DoSomething(i, j);
+            // }
+            var operationsWithDefaultValue = allArgumentOperations
+                .WhereAsArray(argument => argument.ArgumentKind == ArgumentKind.DefaultValue);
+
+            // 5. All the remaining arguments, which might includes method call and a lot of other expressions.
+            // Generate a declaration in the caller.
+            // Example:
+            // Before:
+            // void Caller(bool x)
+            // {
+            //     Callee(Foo(), x ? Foo() : Bar())
+            // }
+            // void Callee(int a, int b)
+            // {
+            //     DoSomething(a, b);
+            // }
+            // After:
+            // void Caller(bool x)
+            // {
+            //     int a = Foo();
+            //     int b = x ? Foo() : Bar();
+            //     DoSomething(a, b);
+            // }
+            // void Callee(int a, int b)
+            // {
+            //     DoSomething(a, b);
+            // }
+            var operationsToGenerateFreshVariablesFor = allArgumentOperations
+                .RemoveRange(operationsWithIdentifierArgument)
+                .RemoveRange(operationsWithVariableDeclarationArgument)
+                .RemoveRange(operationsWithLiteralArgument)
+                .RemoveRange(operationsWithDefaultValue)
+                .WhereAsArray(argument => argument.Children.Single().Syntax is TExpressionSyntax);
```

Comment:
why are you using .Children().Single() instead of just `.Value`?

Reply:
Ah I just realize it has this property points to the Argument :) I will edit this part